[dynamo]Scuba log some debug info about list of integer inputs

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109452
* #109410
* #109411


Some helper functions for jagged tensor have list of integers, which upset dynamic shapes as their content change from run to run. Logging some information to write heuristics.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

anijain2305 committed:
[rfc][dynamo][jagged tensor] Skip quickly instead of waiting for cache size limit

[ghstack-poisoned]

anijain2305 commented:
@ezyang early comments on the direction would be very helpful on this.

ezyang commented on torch/_dynamo/output_graph.py:
```diff
@@ -783,6 +797,9 @@ def compile_subgraph(
         log.debug("COMPILING GRAPH due to %s", reason)
 
+        if skip_reason := self.skip_frame_based_on_heurtisic():
+            raise SkipFrameBasedOnHueristic(skip_reason)
```
Heuristic

ezyang commented:
Two points:

1. If you want to add a heuristic, it is best to have some data on it. Can you first land an instrumentation diff to our Scuba table that collects the necessary info to see if the heuristic would fire or not and on what flows?
2. We generate a lot of guards today, even for things we want to compile. So count_non_getitem_op_calls is really doing a lot of work here. But then my question is, why does it even matter how many guards there are, if there aren't that many operator calls?

anijain2305 committed:
Update on "[rfc][dynamo][jagged tensor] [dynamo] Scuba log some debug info about list of integer inputs"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

anijain2305 commented:
@ezyang sounds good. I have repurposed this PR to log the relevant info list of integers, please take a look.

anijain2305 committed:
Update on "[dynamo]Scuba log some debug info about list of integer inputs"



Some helper functions for jagged tensor have list of integers, which upset dynamic shapes as their content change from run to run. Logging some information to write heuristics.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

ezyang commented on torch/_dynamo/symbolic_convert.py:
```diff
@@ -2084,6 +2086,16 @@ def init_local_index_guards_hack(self):
                 self.output.guards.update(index_guards)
 
+    def collect_debug_info_from_symbolic_locals(self, f_locals):
+        # Count number of SymNodeVariable in symbolic_locals where the value is a list/tuple
```
This seems... insufficient? Like in the KJT case, the list of ints will be nested inside of a UserDefinedObjectVariable

ezyang reviewed (approved):
These fields are very one-offy, which makes me wonder if we should generalize them, or come up with some way we can REMOVE fields in the future, when experiments are done.
